packages/compiler-cli/src/ngtsc/partial_evaluator/src/builtin.ts

```diff
@@ -8,29 +8,45 @@
 
 import * as ts from 'typescript';
 
+import {BuiltinDeclaration} from '../../reflection';
+
 import {DynamicValue} from './dynamic';
 import {BuiltinFn, ResolvedValue, ResolvedValueArray} from './result';
 
+/**
+ * Resolves the specified builtin declaration to a resolved value. For example,
+ * the known JavaScript global `Object` will resolve to a `Map` that provides the
+ * `assign` method with a builtin function. This enables evaluation of `Object.assign`.
+ */
+export function resolveBuiltinDeclaration(builtinDecl: BuiltinDeclaration): ResolvedValue {
+  switch (builtinDecl) {
+    case BuiltinDeclaration.JsGlobalObject:
+      return new Map([['assign', new ObjectAssignBuiltinFn()]]);
+    default:
+      throw new Error(`Cannot resolve unknown builtin declaration ${builtinDecl}`);
```

Comment:
```suggestion
      throw new Error(`Cannot resolve unknown builtin declaration ${BuiltinDeclaration[builtinDecl]}`);
```

(gives the name, rather than the enum index)

Reply:
Good idea. Worth mentioning: I took this from the `TsHelperFn` code 😄 I'll do the same there too.